woocommerce-gateway-stripe.php

```diff
@@ -593,16 +599,44 @@ public function register_routes() {
 					$upe_flag_toggle_controller = new WC_Stripe_REST_UPE_Flag_Toggle_Controller();
 					$upe_flag_toggle_controller->register_routes();
 
-					$gateway = WC()->payment_gateways()->payment_gateways()[ WC_Gateway_Stripe::ID ];
-
 					require_once WC_STRIPE_PLUGIN_PATH . '/includes/admin/class-wc-rest-stripe-settings-controller.php';
-					$settings_controller = new WC_REST_Stripe_Settings_Controller( $gateway );
+					$settings_controller = new WC_REST_Stripe_Settings_Controller( $this->get_main_stripe_gateway() );
 					$settings_controller->register_routes();
 
 					$stripe_account_keys_controller = new WC_REST_Stripe_Account_keys_Controller();
 					$stripe_account_keys_controller->register_routes();
 				}
 			}
+
+			/**
+			 * Returns the main Stripe payment gateway class instance.
+			 *
+			 * @return WC_Stripe_Payment_Gateway
+			 */
+			public function get_main_stripe_gateway() {
```

Comment:
Probably be convenient to make this static. At least other parts or the code are using `WC()->payment_gateways()->payment_gateways()[ WC_Gateway_Stripe::ID ]` if this was static they could use this method too.

Reply:
It's still possible to get this via a static method, like I did in `includes/migrations/class-allowed-payment-request-button-types-update.php`:
```
woocommerce_gateway_stripe()->get_main_stripe_gateway()
```

The problem with static methods is that they're slightly harder to test for.